code/modules/projectiles/guns/energy/laser.dm

```diff
@@ -8,6 +8,33 @@
 	ammo_type = list(/obj/item/ammo_casing/energy/lasergun)
 	ammo_x_offset = 1
 	shaded_charge = 1
+	var/obj/item/external_lens/stored = null
+	modifystate = TRUE
+
+/obj/item/gun/energy/laser/attackby(obj/item/I, mob/user, params)
+	..()
+	if(istype(I, /obj/item/external_lens) && modifystate)
+		if(!stored)
+			equiplens(I,user)

```

Comment:
You should really make the lenses apply a component instead of this

Reply:
@ninjanomnom i made it a component but i m not sure how to remove the ammo type when detaching the lens, as its a list of proj objects and if i qdel the proj it creates a null which breaks selection 
are there things that remove from a list and also fix the list lenght?